mne/viz/evoked.py

```diff
@@ -249,10 +274,20 @@ def _plot_evoked(evoked, picks, exclude, unit, show,
                 gfp_only = (isinstance(gfp, string_types) and gfp == 'only')
                 if not gfp_only:
                     if spatial_colors:
-                        chs = [evoked.info['chs'][i] for i in idx]
+                        chs = [info['chs'][i] for i in idx]
                         locs3d = np.array([ch['loc'][:3] for ch in chs])
                         x, y, z = locs3d.T
                         colors = _rgb(x, y, z)
+                        layout = find_layout(info, ch_type=t, exclude=[])
+                        # drop channels that are not in the data
+                        names = np.intersect1d(layout.names,
+                                               np.array(info['ch_names'])[idx])
+                        name_idx = [layout.names.index(name) for name in names]
+                        # find indices for bads
+                        bads = [np.where(names == bad)[0][0] for bad in
+                                info['bads'] if bad in names]
+                        pos = layout.pos[name_idx, :2]
```

Comment:
this part of the code looks brittle. Maybe a test here? Not sure how you'd do it though ...